### PR TITLE
Add detox test

### PR DESCRIPTION
**Pull request by smrizzo**
### Detox
[Detox Documentation](https://github.com/wix/Detox/blob/master/docs/README.md)
1) Added some login UI testing in a folder named e2e/firstTest.spec.js.
2) This is an example of the structure of creating these UI test.
3) In order to make these test work in that file, need to add testID="" into the props of the specific component on the screen you are testing.
4) Detox only works with react-native components, it will not work with custom made components and a lot of others. 
5) In order to make this work need to pass props down to a component that is from the react-native library. 

### IOS build only right now
1) Clone our project 
2) `npm install`
3) `brew tap wix/brew`
4) `brew install applesimutils`
5) `npm install -g detox-cli`
6) `detox build --configuration ios.sim.debug`
 * If react native packager does not open up then `npm start`
7) `detox test --configuration ios.sim.debug`

This should build your app as if your building it when you type in `react-native run-ios`
Then then after running detox test--configuration ios.sim.debug it will run the test I currently have

### CODE REVIEW

File: AmazonHandmadeRN/src/App.js [hunk 1]
```diff
@@ -85,10 +86,15 @@ class Main extends Component {
           ref={ref => this.dropdown = ref}
         />
       </View>
+      
     )
   }
 }
 
+{/* <TouchableOpacity testID='greetings' onPress={() => this.setState({ greetings: 'new text now' })}>
```
Comment (ianbattin) [inline AmazonHandmadeRN/src/App.js]: Unless this is going to be uncommented in a future commit, delete it for now

File: AmazonHandmadeRN/src/screens/ArtisanHub/ArtisanList.js [hunk 2]
```diff
@@ -44,8 +44,12 @@ export default class ArtisanList extends Component {
     this.props.navigation.navigate('ArtisanDetail', {...artisan})
   }
 
-  _renderArtisanItem = ({item}) => (
+  _renderArtisanItem = ({item, index}) => {
+     console.log(`listItem${index}`);
```
Comment (ianbattin) [inline AmazonHandmadeRN/src/screens/ArtisanHub/ArtisanList.js]: Remove unnecessary console logs

**Review by ianbattin — CHANGES_REQUESTED** [review]

File: AmazonHandmadeRN/src/App.js (hunk 1 above)
Comment (smrizzo) [reply AmazonHandmadeRN/src/App.js]: Okay will remove that then.

File: AmazonHandmadeRN/src/screens/ArtisanHub/ArtisanList.js (hunk 2 above)
Comment (smrizzo) [reply AmazonHandmadeRN/src/screens/ArtisanHub/ArtisanList.js]: Also will remove console log, and be sure to remove those before PR.

File: e2e/init.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+afterEach(async function () {
+  await adapter.afterEach(this);
+});
+
+after(async () => {
+  await detox.cleanup();
+});
+
+// const detox = require('detox');
```
Comment (ianbattin) [inline e2e/init.js]: Is there a reason all this commented out code is here?

File: src/App.js [hunk 4]
```diff
@@ -85,10 +86,15 @@ class Main extends Component {
           ref={ref => this.dropdown = ref}
         />
       </View>
+      
     )
   }
 }
 
+{/* <TouchableOpacity testID='greetings' onPress={() => this.setState({ greetings: 'new text now' })}>
```
Comment (ianbattin) [inline src/App.js]: Delete this

File: src/screens/ArtisanHub/ArtisanList.js [hunk 5]
```diff
@@ -44,8 +44,12 @@ export default class ArtisanList extends Component {
     this.props.navigation.navigate('ArtisanDetail', {...artisan})
   }
 
-  _renderArtisanItem = ({item}) => (
+  _renderArtisanItem = ({item, index}) => {
+     console.log(`listItem${index}`);
```
Comment (ianbattin) [inline src/screens/ArtisanHub/ArtisanList.js]: Delete this

**Review by ianbattin — CHANGES_REQUESTED** [review]

File: e2e/artisanDetail.spec.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
+describe('Artisan list and Details', () => {
+   beforeEach(async () => {
+    
+   });
+
+   it('enter login information and submit', async () => {
+    await element(by.id('email_login')).replaceText('user@user.com');
```
Comment (ianbattin) [inline e2e/artisanDetail.spec.js:7]: Indentation should be 2 spaces not 1

File: e2e/artisanDetail.spec.js [hunk 7]
```diff
@@ -0,0 +1,76 @@
...
+    await element(by.id('artisan_list')).scrollTo('bottom');
+    await expect(element(by.id('listItem5'))).toBeVisible();
+    await element(by.id('listItem5')).tap();
+    await expect(element(by.text('Artisan Details'))).toBeVisible();
+    await expect(element(by.text('ArtisanHub'))).toBeVisible();
+    await element(by.text('ArtisanHub')).tap();
+   })
+
+   it('logout successfully', async () => {
+   await expect(element(by.text('Settings'))).toBeVisible();
```
Comment (ianbattin) [inline e2e/artisanDetail.spec.js]: 2 spaces

File: e2e/firstTest.spec.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
+describe('Login Screen', () => {
+  beforeEach(async () => {
+    //await device.reloadReactNative();
+    //await device.launchApp({newInstance: true});
+  });
+
+  it('HandMade logo is visible', async () => {
+   //await expect(element(by.id('create_account'))).toBeVisible();
+   await expect(element(by.id('handmade_logo'))).toBeVisible();
```
Comment (ianbattin) [inline e2e/firstTest.spec.js]: 2 spaces needed

File: e2e/init.js (hunk 3 above)
Comment (smrizzo) [reply e2e/init.js]: No should be gone now

File: src/App.js (hunk 4 above)
Comment (smrizzo) [reply src/App.js]: Yep deleted

File: src/screens/ArtisanHub/ArtisanList.js (hunk 5 above)
Comment (smrizzo) [reply src/screens/ArtisanHub/ArtisanList.js]: Deleted

**Review by ianbattin — CHANGES_REQUESTED** [review]

**Review by ianbattin — CHANGES_REQUESTED** [review]
Also can you add a section to the README containing your instructions which you have in the PRs comment?

**Comment by smrizzo** [comment]
> Also can you add a section to the README containing your instructions which you have in the PRs comment?

**Comment by smrizzo** [comment]
> Also can you add a section to the README containing your instructions which you have in the PRs comment?

Yeah I can start that after start the code review

**Comment by ianbattin** [comment]
> > Also can you add a section to the README containing your instructions which you have in the PRs comment?
> 
> Yeah I can start that after start the code review

You can just edit the README contained in this branch, that way those changes become a part of this PR, not sure if thats what you meant

**Comment by smrizzo** [comment]
> > > Also can you add a section to the README containing your instructions which you have in the PRs comment?
> > 
> > 
> > Yeah I can start that after start the code review
> 
> You can just edit the README contained in this branch, that way those changes become a part of this PR, not sure if thats what you meant

**Review by ianbattin — APPROVED** [review]
Also can you add a section to the README containing your instructions which you have in the PRs comment?

**Pull request merged**